dev/devicelab/bin/tasks/plugin_lint_mac.dart

```diff
@@ -533,3 +533,22 @@ void _validateMacOSPodfile(String appPath) {
     'macos',
   ));
 }
+
+Future<void> _tryMacOSLint(String executable, List<String> arguments) async {
+  final StringBuffer lintStdout = StringBuffer();
+  try {
+    await eval(
+      executable,
+      arguments,
+      stdout: lintStdout,
+    );
+  } on BuildFailedError {
+    // Temporarily ignore errors due to DT_TOOLCHAIN_DIR. This error was
+    // introduced with Xcode 15. Fix was made in Cocoapods, but is not in an
+    // official release yet.
+    // TODO(vashworth): Stop ignoring when https://github.com/flutter/flutter/issues/133584 is complete.
+    if (!lintStdout.toString().contains('error: DT_TOOLCHAIN_DIR cannot be used to evaluate')) {
```

Comment:
Isn't this going to ignore *all* failures with Xcode 15, because every failure will contain this as one of the lines, regardless of any other failures?

Reply:
Pretty much. Basically `pod lib lint` won't work for macOS until the Cocoapods fix is in an official release and we update our CI to use it. 

We could alternatively check for the Xcode version first and skip linting if it's Xcode 15+.

Reply:
Just made a slight improvement so if there's more errors than just the `DT_TOOLCHAIN_DIR` error, it'll still throw it